### ISSUE TITLE
Integrate Z indexing support for symbols 

### ISSUE THREAD

**Issue by tobrun** [mapbox/mapbox-plugins-android#657]
Upstream in https://github.com/mapbox/mapbox-gl-native/pull/12783, integration for changing order of z index landed for symbolLayer. We need to hook into this configuration and expose a z-index configuration on our end that conforms to the selected configuration.

**Comment by lilykaiser** [comment]
You mean the sort-by-y-position vs sort-by-datasource-order option?

**Comment by tobrun** [comment]
This will be integrated thorugh sort by datasource order (this also means this feature is limited to symbols and available for fill, circle or line). Every time a property is updated we will have the ability to sort the data. 

WIP that depends on v6.6.0 of the maps sdk:
![image 2](https://user-images.githubusercontent.com/2151639/45492607-0485dc80-b76d-11e8-8cb4-4692e1412736.png)

**Comment by tobrun** [comment]
Will ship with v6.6.0, issue on hold with https://github.com/mapbox/mapbox-plugins-android/issues/681

**Comment by tobrun** [comment]
landed with #718
